core/iwasm/compilation/aot_emit_memory.c

```diff
@@ -987,13 +1003,13 @@ aot_compile_op_memory_init(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
 
     POP_I32(len);
     POP_I32(offset);
-    POP_I32(dst);
+    POP_MEM_OFFSET(dst);
 
     param_types[0] = INT8_PTR_TYPE;
     param_types[1] = I32_TYPE;
     param_types[2] = I32_TYPE;
     param_types[3] = I32_TYPE;
-    param_types[4] = I32_TYPE;
+    param_types[4] = MEMORY64_COND_VALUE(I64_TYPE, I32_TYPE);
```

Comment:
The `dst` argument of `aot_memory_init` is `mem_offset_t`, which is defined as:
```C
#if WASM_ENABLE_MEMORY64 != 0
typedef uint64 mem_offset_t;
#else
typedef uint32 mem_offset_t;
#endif
```
So it is uint64 when WASM_ENABLE_MEMORY64 is set as 1, no matter the actual memory is 64 bit or 32 bit.
And here the param type of dst is decided according to the memory type, it is I32_TYPE when the memory is 32 bit. The issue reported in #3375 may occur again: aot runtime enables memory64 feature and the argument type is uint64, while memory in aot file is 32bit and aot code passes uint32 to the argument.

How about changing the param type in aot_memory_init to size_t, and here set the param type according to `comp_ctx->pointer_size`, and extend the param to uint64 if needed before passing to aot_memory_init? Similar to #3378.